### ISSUE TITLE
Smoothing and additive blending

### ISSUE THREAD

**Issue by sebleedelisle** [openframeworks/openFrameworks#1546]
Hi guys! 

I'm not sure why but as soon as you enable smoothing, additive blending is disabled. In fact I think someone on the team is also unsure - see the comment "why do we need this?" in libs/openFrameworks/gl/ofGLRenderer.cpp :) 

``` c++
//----------------------------------------------------------
void ofGLRenderer::startSmoothing(){
    #ifndef TARGET_OPENGLES
        glPushAttrib(GL_COLOR_BUFFER_BIT | GL_ENABLE_BIT);
    #endif

    glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
    glEnable(GL_LINE_SMOOTH);

    //why do we need this?
    glEnable(GL_BLEND);
    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
}
```

I commented it out and it seemed to work OK, but I'm sure there are other repercussions. But it would be nice to have smoothing and blendmodes. 

cheers
Seb

**Comment by armadillu** [comment]
Yep I just wasted quite a few hours on this; if ofEnableSmoothing() has been called at any time, every draw call (to ofMesh, etc) re-sets the blending mode. This is particularly painful if you are doing pingpong shaders as every fbo->draw() re-enables the blending and gives very bizarre results.
